### PR TITLE
#218 delete lost issues when we can't find them

### LINKED ISSUE #218: delete lost issues when we can't find them
This is what I'm getting from `label-was-attached.rb`: 

```
GET https://api.github.com/repositories/728758275/issues/161/timeline?per_page=100: 404 - Not Found // See: https://docs.github.com/rest/issues/timeline#list-timeline-events-for-an-issue
```

This means that the issue is gone (transferred to another repository). I believe, we should catch this type of error and remove the `issue-was-opened` fact from the factbase.

### PR DESCRIPTION
**Pull request by Suban05**
## Changes

* Added removing the issue when the GitHub API returns a 404 status
* Added tests

Fixes #218

### CODE REVIEW

**Comment by Suban05** [comment]
@yegor256 can you please check if I am using the correct factbase for deleting a lost issue?

File: judges/label-was-attached/label-was-attached.rb [hunk 1]
```diff
@@ -34,6 +34,7 @@
   repeats 20
   over do |repository, issue|
     Fbe.octo.issue_timeline(repository, issue).each do |te|
+      Fbe.fb.query("(eq issue #{issue})").delete! if te[:status] == '404'
```
Comment (yegor256) [inline judges/label-was-attached/label-was-attached.rb:37]: @Suban05 all correct, but you will delete facts related to other repositories too. Better do this:

```
(and (eq where 'github') (eq repository #{repository}) (eq issue #{issue}))
```

**Comment by yegor256** [comment]
@Suban05 all good, but small comment above

File: judges/label-was-attached/label-was-attached.rb (hunk 1 above)
Comment (Suban05) [reply judges/label-was-attached/label-was-attached.rb]: @yegor256 thanks! Also, I have added a test of this case

**Comment by Suban05** [comment]
@yegor256 fixed

**Comment by Suban05** [comment]
@Yegorov can you review it, please?

**Comment by Yegorov** [comment]
@Suban05 Looks great, no complaints!

**Comment by yegor256** [comment]
@Yegorov you have to make a "review" (click "submit review"), otherwise it won't count

**Comment by yegor256** [comment]
@Yegorov like this:

Screenshot 2024-08-04 at 12 42 13

**Review by Yegorov — APPROVED** [review]
Looks great, no complaints!

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@Suban05 thanks!

**Comment by 0crat** [comment]
@yegor256 Hey there, code wizard! 🧙‍♂️ Awesome job on that review! You've snagged +24 points for your effort: +25 for stepping up, +9 for tackling those 93 hits-of-code (nice!), but we had to dock -10 because there were only 3 comments. Remember, more feedback is always better! Your current score sits at [-535](https://zerocracy.github.io/judges-action/zerocracy-vitals.html). Keep up the great work and let's aim for even more engaging reviews next time!

**Comment by 0crat** [comment]
@Suban05 Awesome work! 🎉 You just scored +46 points (+30 base + 16 for 167 hits-of-code). Your total's now [+152](https://zerocracy.github.io/judges-action/zerocracy-vitals.html). [Keep the momentum going](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) - we love seeing your contributions!

**Comment by 0crat** [comment]
@Suban05 Thanks for your review! Here's the breakdown of your +5 points: +25 base points, -40 for self-review (not recommended), +0.02 points per hit-of-code (capped at 30), +1 point per comment (max 20), adjusted to the minimum 5 points. Remember, reviewing others' code often yields higher rewards. Your current balance is [+302](https://zerocracy.github.io/judges-action/zerocracy-vitals.html). Keep up the good work! 💻

**Comment by 0crat** [comment]
@Yegorov Hey there! 👋 Awesome job on the review! You've snagged +36 points for this one: +25 for being an author, and +11 for those 11 insightful comments you made. 💬 That's pretty good, considering we cap comment points at 20. Your [running balance](https://zerocracy.github.io/judges-action/zerocracy-vitals.html) is now at a solid +194. Keep up the great work, and remember, more [hits-of-code](https://www.yegor256.com/2014/11/14/hits-of-code.html) and comments could bump up your points even more next time!

**Pull request merged**